Add distinct name for WireGuard connections generated by the agent

amalkh5 opened the pull request:
Fixes #465 

it will add the word "agent" to WireGuard connections generated by the agent, for example:
``` shell 
#  host-native WireGuard connections
$ interactive-amals-MacBook-amalkms5-gmail-com-724    

# agent WireGuard connections
$ agent-interactive-amals-MacBook-amalkms5-gmail-com-714
```

amalkh5 commented:
As for the failed checks, it was regreding existing code and I do not know if I should change it or keep it?

rugwirobaker reviewed (changes requested):
Sorry for the late review. Everything looks except course the little deep source complaints. You can change those.

amalkh5 commented:
@rugwirobaker sorry I clicked on review by miskate, I will let you know when I fix the issues

amalkh5 commented:
@rugwirobaker 
I fixed all things suggested in deepsource except the one where it requests to change `math/rand` to `crypto/rand` because I [read](https://yourbasic.org/golang/crypto-rand-int/) that using `crypto/rand. Reader` returns an error if the underlying system call fails.
 For instance, if it can't read /dev/urandom on a Unix system, or if CryptAcquireContext fails on a Windows system.

What do you think?

jsierles commented:
Hey, would you be interested in finishing this up?

amalkh5 commented:
@jsierles sure, I will resolve the conflict and push it.

amalkh5 commented:
@jsierles  Fixed, I removed "interactive" and added the word "agent" to WireGuard connections generated and by the agent, for example:


```
# host-native WireGuard connections
$ interactive-amals-MacBook-xxx-amalxxxx-gmail-com-860    

# agent WireGuard connections
$ agent-amals-MacBook-xxx-amalxxxx-gmail-com-534
```

jsierles commented on internal/wireguard/wg.go:
```diff
@@ -24,6 +24,23 @@ var (
 	cleanDNSPattern = regexp.MustCompile(`[^a-zA-Z0-9\\-]`)
 )
 
+func gernateConnectionnName(apiClient *api.Client, ctx context.Context) (string, error) {
```
This should be named correctly, perhaps with a better name like `generatePeerName`

jsierles replied:
Also, the context should be the first argument passed in.

amalkh5 commented:
@jsierles fixed

jsierles commented:
Thanks! Actually we need to keep the name `interactive-`. Can you reverse it so we have something like `interactive-agent-*`?

jsierles reviewed (commented):
Looks good except for the comment about keeping the `interactive-*` prefix.
